website/versioned_docs/version-0.62/e2e-testing.md

```diff
@@ -0,0 +1,442 @@
+
+```
+browser.setTimeout({ 'implicit': 5000 });
+```
+
+3. waitForPageTimeout.
+
+This timer is provided by BasePage object.
+
+```
+  // Default timeout for waitForPageLoaded command in PageObject
+  privatewaitforPageTimeout: number = 10000;
+```
+
+You can override the setting in test case by passing a timeout parameter when call `waitForPageLoaded`, for example:
+
+```
+LoginPage.waitForPageLoaded(15000)
+```
+
+## wdio.conf.js
+
+### Capabilities
+
+capabilities are the configuration which appium/WinAppDriver used to identify the app and launch the app. Below configurations supports both with and without appium in the setup.
+
+**&#39;appium:app&#39;** is used by appium when the setup is WebDriverIO <-> Appium <-> WinAppDriver,
+
+**&#39;app&#39;** is used for directly connection between WebDriverIO <-> WinAppDriver.
+
+```
+  capabilities: [
+  {
+      maxInstances: 1,
+      platformName: 'windows',
+      'appium:deviceName': 'WindowsPC',
+      'appium:app': 'ReactUWPTestApp_kc2bncckyf4ap!App',
+      'deviceName': 'WindowsPC',
+      'app': 'ReactUWPTestApp_kc2bncckyf4ap!App',
+      'winAppDriver:experimental-w3c': true,
+  },
+
+```
+
+### LogLevel
+```
+    logLevel: 'trace'
+```
+### Timeout
+
+```
+    // Default timeout for all waitFor* commands.
+    waitforTimeout: 10000,
+```
+
+### appium service
+
+Below configuration lets the framework launch/kill appium automatically during the testing, and logs are saved as `reports\appium.txt`.
+
+```
+    port: 4723,
+    services: ['appium'],
+    appium: {
+        logPath: './reports/',
+        args: {
+            port: '4723',
+        }
+    },
+
+```
+
+But the log like below is not that readable because it has a lot of control characters which is used by terminal.
+
+ ![Bad Readability](assets/Bad-Readability.png)
+
+You could get nice output if you launch the appium by yourself:
+
+1. Modify the configuration and remove `appium` from services
+```
+services: [],
+```
+
+2. Start appium by yourself
+```
+.\node\_modules\.bin\appium
+```
+
+### Test framework
+```
+framework: 'jasmine'
+```
+
+### Reports
+
+```
+reporters: ['dot', ['junit', { outputDir : '.\\reports' }]],
+```
+
+## PageObject Pattern
+
+PageObject Pattern is recommended in the E2E test, and you can get more information from these links:
+
+[https://github.com/SeleniumHQ/selenium/wiki/PageObjects](https://github.com/SeleniumHQ/selenium/wiki/PageObjects)
+
+[https://webdriver.io/docs/pageobjects.html](https://webdriver.io/docs/pageobjects.html)
+
+## Element Locator `By`
+
+```
+private get submitButton() {
+    return By('Submit');
+}
+
+this.submitButton.click();
+```
+
+You can easily to use By(string) to locate a element which associated with testID in the app.
+
+It's recommended to define a `get` for each locator like above.
```

Comment:
this file is out of date - we should just link to the repo (under /docs/e2e-testing.md)

Reply:
I'm going to track linking to the main repo under issue #149